backend/controllers/addEmailDomain.test.js

```diff
@@ -0,0 +1,125 @@
+const { createEmailDomainFactory } = require('./addEmailDomain')
+
+describe('create email domain factory', () => {
+  const createEmailDomainApi = jest.fn()
+  const emailDomainFactory = createEmailDomainFactory(createEmailDomainApi, '/create-email-domain', '/email-domains')
+  describe('index', () => {
+    it('should call create email domain render', async () => {
+      const standardReq = {
+        params: {},
+        flash: jest.fn(),
+      }
+      const req = { flash: jest.fn() }
+      const flashDomain = req.flash('domain')
+      const domain = flashDomain != null && flashDomain.length > 0 ? flashDomain[0] : ''
+      const render = jest.fn()
+      await emailDomainFactory.index(standardReq, { render })
+      expect(render).toBeCalledWith('createEmailDomain.njk', {
+        createEmailDomainUrl: '/create-email-domain',
+        listEmailDomainUrl: '/email-domains',
+        ...domain,
+        errors: req.flash('createEmailDomainErrors'),
```

Comment:
Shouldn't the errors be undefined at this point?

Reply:
Yes, req.flash('xxx') is undefined and can put undefined for clarity.